src/cpp/r/RSexp.cpp

```diff
@@ -105,7 +122,14 @@ SEXP findNamespace(const std::string& name)
    // case 4071: namespace look up executes R code that can trip the debugger
    DisableDebugScope disableStepInto(R_GlobalEnv);
 
-   return R_FindNamespace(Rf_mkString(name.c_str()));
+   // R_FindNamespace will throw if it fails to find a particular name.
+   // Instead, we manually search the namespace registry.
+   r::sexp::Protect protect;
+   SEXP nameSEXP;
+   protect.add(nameSEXP = Rf_install(name.c_str()));
```

Comment:
I could be wrong but I don't think you need to protect the results of Rf_install (since the returned SEXP will always end up in the global string table). If you look at uses of Rf_install in the R source code you should see this.